File: Phishing_defencer/popup.py
from flask import Flask, request, jsonify, render_template
from bs4 import BeautifulSoup
import requests
import sys
import pandas as pd
import subprocess
from model import model_RF
from testing_model import test_fnx

app = Flask(__name__)
def index():
    return render_template('index.html')


@app.route('/scrape', methods=['POST'])
def scrape():
    current_url = request.form['currentUrl']
    page = requests.get(current_url)
    soup = BeautifulSoup(page.content, 'html.parser')
    links_list = soup.find_all('a')
    
    scraped_data = []
    
    for link in links_list:
        if 'href' in link.attrs:
            web_link = str(link.attrs['href'])
            a=test_fnx(web_link)
            scraped_data.append({'url': web_link, 'status': a})
    return jsonify(scraped_data) 
            

if __name__ == '__main__':
    app.run(host='127.0.0.1', port=5000)

File: Phishing_defencer/popup.js
document.addEventListener('DOMContentLoaded', function () {
    // Get the button element
    var button = document.getElementById('myButton');

    // Add a click event listener to the button
    button.addEventListener('click', function () {
        // Get the current tab
        chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
            // Get the URL of the current tab
            var currentUrl = tabs[0].url;

            // Make an AJAX request to your Python Flask server
            var xhr = new XMLHttpRequest();
            xhr.open('POST', 'http://localhost:5000/scrape');
            xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
            xhr.onreadystatechange = function () {
                if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
                    // Parse the JSON response
                    var scrapedData = JSON.parse(xhr.responseText);



                    
                    
                    var linksString = 'Scraped Links:\n\n';
                    
                    scrapedData.forEach(function (data) {
                        linksString += data.url + ': ' + data.status + '\n';
                    });
                    
                    // Show an alert with the links
                    alert(linksString);
                }
            };
            xhr.send('currentUrl=' + currentUrl);
        });
    });
    // #### open a new window in full web page
    // // Create a new HTML document
    // var newDoc = document.implementation.createHTMLDocument('Scraped Links');

    // // Add a header to the new document
    // var header = newDoc.createElement('h1');
    // header.textContent = 'Scraped Links:';
    // newDoc.body.appendChild(header);

    // // Add each scraped link to the new document
    // scrapedData.forEach(function (data) {
    //     var linkElement = newDoc.createElement('p');    
    //     linkElement.textContent = data.url + ': ' + data.status;
    //     newDoc.body.appendChild(linkElement);
    // });

    // // Convert the new document to HTML string
    // var htmlString = new XMLSerializer().serializeToString(newDoc);

    // // Open a new tab with the HTML content
    // var newTab = window.open();
    // newTab.document.write(htmlString);









    // ####  Opens a new tab in window
    // var htmlString = '<h1>Scraped Links:</h1>';

    // scrapedData.forEach(function (data) {
    //     htmlString += '<p>' + data.url + ': ' + data.status + '</p>';
    // });

    // // Open a new popup window with the HTML content
    // var popup = window.open('', '_blank', 'width=600,height=400');
    // popup.document.write(htmlString);
});